python-avd/pyavd/_eos_designs/structured_config/underlay/utils.py

```diff
@@ -141,67 +143,60 @@ def _underlay_vlan_trunk_groups(self: AvdStructuredConfigUnderlay) -> list:
     def _uplinks(self: AvdStructuredConfigUnderlay) -> list:
         return get(self._hostvars, "switch.uplinks")
 
-    def _get_l3_interface_cfg(self: AvdStructuredConfigUnderlay, l3_interface: dict) -> dict | None:
+    def _get_l3_interface_cfg(
+        self: AvdStructuredConfigUnderlay, l3_interface: EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3InterfacesItem
+    ) -> dict | None:
         """Returns structured_configuration for one L3 interface."""
-        interface_name = get(l3_interface, "name", required=True, org_key=f"<node_type_key>...[node={self.shared_utils.hostname}].l3_interfaces[].name]")
-
-        interface_description = l3_interface.get("description")
+        interface_description = l3_interface.description
         if not interface_description:
             interface_description = self.shared_utils.interface_descriptions.underlay_ethernet_interface(
                 InterfaceDescriptionData(
                     shared_utils=self.shared_utils,
-                    interface=interface_name,
-                    peer=l3_interface.get("peer"),
-                    peer_interface=l3_interface.get("peer_interface"),
-                    wan_carrier=l3_interface.get("wan_carrier"),
-                    wan_circuit_id=l3_interface.get("wan_circuit_id"),
+                    interface=l3_interface.name,
+                    peer=l3_interface.peer,
+                    peer_interface=l3_interface.peer_interface,
+                    wan_carrier=l3_interface.wan_carrier,
+                    wan_circuit_id=l3_interface.wan_circuit_id,
                 ),
             )
 
         # TODO: catch if ip_address is not valid or not dhcp
-        ip_address = get(
-            l3_interface,
-            "ip_address",
-            required=True,
-            org_key=f"{self.shared_utils.node_type_key_data['key']}.nodes[name={self.shared_utils.hostname}].l3_interfaces[name={interface_name}].ip_address",
-        )
-
         interface = {
-            "name": interface_name,
+            "name": l3_interface.name,
             "peer_type": "l3_interface",
-            "peer": l3_interface.get("peer"),
-            "peer_interface": l3_interface.get("peer_interface"),
-            "ip_address": ip_address,
-            "shutdown": not l3_interface.get("enabled", True),
-            "switchport": {"enabled": False if "." not in interface_name else None},
+            "peer": l3_interface.peer,
+            "peer_interface": l3_interface.peer_interface,
+            "ip_address": l3_interface.ip_address,
```

Comment:
technically `ip_address` is not mandatory in the schema for `l3_interfaces` so we are losing a check (line 162)

Reply:
If we wish to enforce it, we should make it required in the schema, but soon we will have ipv6 etc so we cannot require ipv4 ip. I will add a check and raise missing variable so same as before.

Reply:
done